tgui/packages/tgui/interfaces/RobotSelfDiagnosis.js

```diff
@@ -0,0 +1,87 @@
+import { useBackend } from "../backend";
+import { NoticeBox, Flex, LabeledList, Section } from "../components";
+import { Window } from "../layouts";
+import { capitalize } from 'common/string';
+
+export const RobotSelfDiagnosis = (props, context) => {
+  const { data } = useBackend(context);
+  const { component_data } = data;
+  const getDamageColor = (damage, maxDamage) => {
+    let damageRatio = damage / maxDamage;
+    if (damageRatio <= 0.20) {
+      return "good";
+    }
+    else if (damageRatio <= 0.50) {
+      return "average";
+    }
+    else {
+      return "bad";
+    }
+  };
+  const getStatusColor = status => {
+    return status ? "good" : "bad";
+  };
+  return (
+    <Window>
+      <Window.Content scrollable>
+        {component_data.map((entry, i) => (
+          <Section
+            key={i}
+            open
+            title={capitalize(entry.name)}>
+            {entry.installed <= 0 ? (
+              <NoticeBox
+                m={-0.5}
+                height={3.5}
+                color="red"
+                style={{
+                  "font-style": "normal",
+                }}>
+                <Flex height="100%">
+                  <Flex.Item
+                    grow="1"
+                    textAlign="center"
```

Comment:
Since it's a number, I believe this is preferred:
```suggestion
                  <Flex.Item
                    grow={1}
                    textAlign="center"
```

Reply:
Doesn't really matter IMO

Reply:
Definitely more of a stylistic thing than anything actually code related, but I've seen more numbers in brackets than in quotes in TGUI code, so that's what I would go for.

Reply:
Doesn't really matter yeah, but I'll standardize on using the `{1}`.